src/fluent-bit.c

```diff
@@ -1425,42 +1429,47 @@ int flb_main(int argc, char **argv)
     }
 #endif
 
-    while (ctx->status == FLB_LIB_OK && exit_signal == 0) {
-        sleep(1);
-
-#ifdef FLB_SYSTEM_WINDOWS
-        if (handler_signal == 1) {
-            handler_signal = 0;
-            flb_reload(ctx, cf_opts);
-        }
-        else if (handler_signal == 2){
-            handler_signal = 0;
-            break;
-        }
-#endif
-
-        /* set the context again before checking the status again */
-        ctx = flb_context_get();
+    while (exit_signal == 0 && is_shutdown == 0) {
```

Comment:
Did you add the appropriate code to detect a reload and exit the `mk_event_foreach` scope that I told you about or not? I think if you did I might've missed it when reading the code.

Reply:
I have actually been thinking about how to rewrite that code as cleanly as possible, ie:

- abstract out the decision into a small internal function
- use ifs instead of a switch case
- not rewrite it all and add another is_reload variable to flag escaping the loop

Reply:
I decided upon adding a `is_reload` flag. 

In the future the `ctx` or `config` object really should be separated and one made to be wholly owned by `flb_main`. For now this is the best I can do at the moment without a larger refactor.